oauthex/oauthex_test.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2025 The Go MCP SDK Authors. All rights reserved.
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file.
+
+//go:build mcp_go_client_oauth
+
+package oauthex
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
+)
+
+const (
+	authServerPort = ":8080"
+	issuer         = "http://localhost" + authServerPort
```

Comment:
I feel like we should probably close over this information.

Does the server have to run on :8080 in order to function? If so, that's a problem, because two instances of this server can't run concurrently. Better to use the httptest server URL, if possible.